cpp/src/arrow/engine/substrait/extension_set.cc

```diff
@@ -288,6 +808,11 @@ struct ExtensionIdRegistryImpl : ExtensionIdRegistry {
     return Status::OK();
   }
 
+  Status RegisterFunctionMapping(Id id, SubstraitToArrow conversion_func) override {
+    DCHECK_OK(functions_map.AddSubstraitToArrow(id.name.to_string(), conversion_func));  
+    return RegisterFunction(id, id.name.to_string());
```

Comment:
It seems a little odd that we need two maps.  What happens if two functions exist with the same name but different URIs?  Thinking on this longer, maybe `substrait_to_arrow` should replace the map in the extension id registry (that gets updated by the call to `RegisterFunction`?)